cmd/zoekt-archive-index/ignore.go

```diff
@@ -0,0 +1,79 @@
+package main
+
+import (
+	"bufio"
+	"io"
+	"strings"
+)
+
+var (
+	lineComment = "#"
+	ignoreFile  = ".sourcegraph/sourcegraphignore"
+)
+
+type IgnoreMatcher struct {
```

Comment:
I export `IgnoreMatcher`, `ParseSourcegraphignore`, and `Match` because we can reuse them in `searcher`.